site/topics/particle-swarm-optimization/pso.rst

```diff
@@ -106,33 +106,134 @@ Velocity Calculation
+    * :math:`\vec{p_{i}}_{best}` is the particle's best known position within the search space
+    * :math:`\vec{x_{i}}(t)` is the particle's position at time :math:`t`
+
+
+* The difference between the particle's best known position and current position dictates where the particle needs to go
+
+    * :math:`\vec{p_{i}}_{best} - \vec{x_{i}}(t)`
+
+
+* :math:`c_{1}` and :math:`r_{1}` scale the vector
+
+
+.. figure:: plot_best_vs_current_position.png
+    :width: 500 px
+    :align: center
+
+    Vector (blue) showing the difference between the particle's best known position (red) and its current position
+    (green). The vector :math:`(-1, -3)` is shown starting at the current position :math:`(3, 4)`. If the particle
+    were to have exactly this velocity for one time step, it would return to the best known position.
+
 
 Social Term: :math:`c_{2}\vec{r_{2}}(\vec{g}_{best} - \vec{x_{i}}(t))`
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+* Similarly, each particle is influenced by the population's best known position
+
+    * :math:`c_{2}\vec{r_{2}}(\vec{g}_{best} - \vec{x_{i}}(t))`
+
+* Where
+
+    * :math:`i` is some particle
+    * :math:`c_{2}` is some coefficient used to control how much the particle is influenced by the population's best
+
+        * :math:`c_{2} \in [0, 2]`
+
+
+    * :math:`r_{2}` is some stochastic vector discussed below
+    * :math:`\vec{g}_{best}` is the population's best known position within the search space
+    * :math:`\vec{x_{i}}(t)` is the particle's position at time :math:`t`
+
+
 
 Random/Stochastic Components: :math:`\vec{r_{1}}` and :math:`\vec{r_{2}}`
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+* The cognitive and social portions of the velocity update included the vectors :math:`\vec{r_{1}}` and :math:`\vec{r_{2}}` respectively
+* These vectors have values between :math:`[0, 1]` that are stochastically determined
+
+    * Randomly determined for each velocity update calculation for each particle
+
+
+* These random/stochastic vectors are important as they add a chance for novelty
+* Further, they have been empirically shown to improve the search and prevent premature convergence
+
+
+Putting the Velocity Update Together
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* The velocity update is the sum of the parts of the update
+
+    * Inertia term + cognitive term + social term
+
+
+* Velocity update for some particle :math:`i`
+
+.. math::
+
+    \vec{v_{i}}(t+1) = \omega\vec{v_{i}}(t)
+        + c_{1}\vec{r_{1}}(\vec{p_{i}}_{best} - \vec{x_{i}}(t))
+        + c_{2}\vec{r_{2}}(\vec{g}_{best} - \vec{x_{i}}(t))
+
+
+* As discussed, there are three coefficients that need to be tuned for the algorithm
+* As a starting place, `van den Bergh <https://repository.up.ac.za/bitstream/handle/2263/24297/00thesis.pdf?sequence=1>`_ suggests
+
+    * :math:`\omega = 0.729844`
+    * :math:`c_{1} = c_{2} = 1.496180`
```

Comment:
Weird

Reply:
Agreed. It's just some dudes thesis, but everyone seems to ref it.